plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go

```diff
@@ -0,0 +1,244 @@
+		So the only way is to hardcode reserved IP (https://en.wikipedia.org/wiki/Link-local_address)
+	*/
+	//nolint:gosec // G101: Potential hardcoded credentials - false positive
+	defaultMetadataTokenURL  = "http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token"
+	defaultMetadataFolderURL = "http://169.254.169.254/computeMetadata/v1/yandex/folder-id"
+)
+
+func (*NebiusCloudMonitoring) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *NebiusCloudMonitoring) Init() error {
+	if a.Timeout <= 0 {
+		a.Timeout = config.Duration(defaultRequestTimeout)
+	}
+	if a.Endpoint == "" {
+		a.Endpoint = defaultEndpoint
+	}
+	if a.Service == "" {
+		a.Service = "custom"
+	}
+	if a.metadataTokenURL == "" {
+		a.metadataTokenURL = defaultMetadataTokenURL
+	}
+	if a.metadataFolderURL == "" {
+		a.metadataFolderURL = defaultMetadataFolderURL
+	}
+
+	a.client = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: time.Duration(a.Timeout),
+	}
+	tags := map[string]string{}
+	a.MetricOutsideWindow = selfstat.Register("nebius_cloud_monitoring", "metric_outside_window", tags)
+	return nil
+}
+
+// Connect initializes the plugin and validates connectivity
+func (a *NebiusCloudMonitoring) Connect() error {
+	a.Log.Infof("getting folder ID in %s", a.metadataFolderURL)
+	body, err := a.getResponseFromMetadata(a.client, a.metadataFolderURL)
+	if err != nil {
+		return err
+	}
+	a.folderID = string(body)
+	if a.folderID == "" {
+		return fmt.Errorf("unable to fetch folder id from URL %s: %w", a.metadataFolderURL, err)
+	}
+	a.Log.Infof("Writing to Nebius.Cloud Monitoring URL: %s", a.Endpoint)
+
+	return nil
+}
+
+// Close shuts down an any active connections
+func (a *NebiusCloudMonitoring) Close() error {
+	a.client = nil
+	return nil
+}
+
+// Write writes metrics to the remote endpoint
+func (a *NebiusCloudMonitoring) Write(metrics []telegraf.Metric) error {
+	var nebiusCloudMonitoringMetrics []nebiusCloudMonitoringMetric
+	for _, m := range metrics {
+		for _, field := range m.FieldList() {
+			value, err := internal.ToFloat64(field.Value)
+			if err != nil {
+				a.Log.Errorf("skipping value: %w", err.Error())
+				continue
+			}
+
+			nebiusCloudMonitoringMetrics = append(
+				nebiusCloudMonitoringMetrics,
+				nebiusCloudMonitoringMetric{
+					Name:   m.Name() + "_" + field.Key,
+					Labels: m.Tags(),
+					TS:     m.Time().Format(time.RFC3339),
+					Value:  value,
+				},
+			)
+		}
+	}
+
+	body, err := json.Marshal(
+		nebiusCloudMonitoringMessage{
+			Metrics: nebiusCloudMonitoringMetrics,
+		},
+	)
+	if err != nil {
+		return err
+	}
+	body = append(body, '\n')
+	return a.send(body)
+}
+
+func (a *NebiusCloudMonitoring) getResponseFromMetadata(c *http.Client, metadataURL string) ([]byte, error) {
+	req, err := http.NewRequest("GET", metadataURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+	req.Header.Set("Metadata-Flavor", "Google")
```

Comment:
I see this is required for Google, how does AWS react to this? Does it just ignore it?

Reply:
On Nebius Cloud Platform both of methods of getting a metadata implemented (through google like and aws like style). I decided to do it through gcp style.

Reply:
Ah thank you for the explination